src/ufo.rs

```diff
@@ -271,27 +233,15 @@ impl Font {
             };
 
             let glyph_names = NameList::default();
-            let layers: Vec<LayerInfo> = if self.data_request.layers {
-                let mut contents = match meta.format_version {
-                    FormatVersion::V3 => {
-                        let contents_path = path.join(LAYER_CONTENTS_FILE);
-                        let contents: Vec<(String, PathBuf)> = plist::from_file(contents_path)?;
-                        contents
-                    }
-                    _older => vec![(DEFAULT_LAYER_NAME.into(), DEFAULT_GLYPHS_DIRNAME.into())],
-                };
-
-                let layers_r: Result<Vec<LayerInfo>, Error> = contents
-                    .drain(..)
-                    .map(|(name, p)| {
-                        let layer_path = path.join(&p);
-                        let layer = Layer::load_impl(&layer_path, &glyph_names)?;
-                        Ok(LayerInfo { name, path: p, layer })
-                    })
-                    .collect();
-                layers_r?
+            let layers = if self.data_request.layers {
+                if matches!(meta.format_version, FormatVersion::V3)
```

Comment:
Why not `if meta.format_version == FormatVersion::V3 && ...`?

Reply:
no compelling reason!